tests/gis_tests/gis_migrations/test_operations.py

```diff
@@ -214,6 +228,98 @@ def test_create_model_spatial_index(self):
         if connection.features.supports_raster:
             self.assertSpatialIndexExists("gis_neighborhood", "rast", raster=True)
 
+    @skipUnlessDBFeature("can_alter_geometry_field")
+    def test_alter_field_remove_spatial_index(self):
+        """
+        Test that indexes are removed when `spatial_index` is changed to False
+        """
+        if not self.has_spatial_indexes:
+            self.skipTest("No support for Spatial indexes")
+
+        self.alter_gis_model(
+            migrations.AlterField,
+            "Neighborhood",
+            "geom",
+            False,
+            fields.MultiPolygonField,
+            field_class_kwargs={"srid": 4326, "spatial_index": False},
+        )
+        self.assertSpatialIndexNotExists("gis_neighborhood", "geom")
+
+    @skipUnlessDBFeature("can_alter_geometry_field")
+    def test_alter_field_add_spatial_index(self):
+        """
+        Test that indexes are added when `spatial_index` is changed to True
+        """
+        if not self.has_spatial_indexes:
+            self.skipTest("No support for Spatial indexes")
+
+        self.alter_gis_model(
+            migrations.AddField,
+            "Neighborhood",
+            "point",
+            False,
+            fields.PointField,
+            field_class_kwargs={"srid": 4326, "spatial_index": False},
+        )
+        self.assertSpatialIndexNotExists("gis_neighborhood", "point")
+
+        self.alter_gis_model(
+            migrations.AlterField,
+            "Neighborhood",
+            "point",
+            False,
+            fields.PointField,
+            field_class_kwargs={"srid": 4326, "spatial_index": True},
+        )
+        self.assertSpatialIndexExists("gis_neighborhood", "point")
+
+    @skipUnlessDBFeature("supports_raster", "can_alter_geometry_field")
+    def test_alter_raster_field_remove_spatial_index(self):
+        """
+        Test that raster indexes are removed when `spatial_index` is changed to False
+        """
+        if not self.has_spatial_indexes:
+            self.skipTest("No support for Spatial indexes")
+
+        self.alter_gis_model(
+            migrations.AlterField,
+            "Neighborhood",
+            "rast",
+            False,
+            fields.RasterField,
+            field_class_kwargs={"srid": 4326, "null": True, "spatial_index": False},
+        )
+        self.assertSpatialIndexNotExists("gis_neighborhood", "rast", raster=True)
+
+    @skipUnlessDBFeature("supports_raster", "can_alter_geometry_field")
+    def test_alter_raster_field_add_spatial_index(self):
```

Comment:
Is there any specific here for `RasterField`? If not, I'd remove `test_alter_raster_field_remove_spatial_index` and `test_alter_raster_field_add_spatial_index` as they are redundant.

Reply:
Looks like only PostGIS supports raster fields and there is no difference there, so I'll remove the raster tests.